tests/integration/util/lib.go

```diff
@@ -0,0 +1,418 @@
+func DeleteNamespace(t *testing.T, helmOptions *helm.Options) bool {
+
+	namespace := helmOptions.KubectlOptions.Namespace
+	deleteErr := k8s.RunKubectlE(t, helmOptions.KubectlOptions, "delete", "namespace", namespace)
+	Log(t, "Delete", "Namespace", fmt.Sprintf("Using ns: %s", namespace), deleteErr)
+	return deleteErr == nil
+}
+
+// DeleteOperatorCRD performs lookup of resource targets specific to CRD resource name.
+// Removes the dependency resources and finalizers
+func DeleteOperatorCRD(t *testing.T, kubeOptions *k8s.KubectlOptions, crdIdentity CRDIdentity, timeoutSeconds int) bool {
+
+	Log(t, "\n===== DELETE CRD", "", "", nil)
+	Log(t, "DeleteCRD", "lookup CRD by metaName: ", crdIdentity.CRDResourceName, nil)
+	if lookupCRDByMetaName(t, kubeOptions, crdIdentity.CRDResourceName) {
+
+		errResults := deleteChildResources(t, kubeOptions, getChildResources(), crdIdentity.CRDHookName)
+		if len(errResults) != 0 {
+			LogResults(t, "Delete CRD child resource removal failures.", errResults)
+			return false
+		}
+
+		isFinalizersRemoved := removeFinalizers(t, kubeOptions, crdIdentity.CRDName)
+		if !isFinalizersRemoved {
+			Log(t, "Finalizers NOT removed.", "removeFinalizers", "", nil)
+			return false
+		}
+
+		isCRDDeleted := deleteCustomResourceDefinition(t, kubeOptions, crdIdentity.CRDName, timeoutSeconds)
+		if !isCRDDeleted {
+			Log(t, "CRD NOT deleted.", "deleteCustomResourceDefinition", "", nil)
+			return false
+		}
+	}
+	return true
+}
+
+// DeleteClusterCRD removes the k8ssandra c* cluster CRD
+func DeleteClusterCRD(t *testing.T, kubeOptions *k8s.KubectlOptions, crdIdentity ClusterIdentity, timeoutSeconds int) bool {
+
+	Log(t, "\n===== DELETE CRD", "", "", nil)
+	Log(t, "DeleteCRD", "lookup CRD by metaName: ", crdIdentity.ClusterName, nil)
+	if lookupCRDByMetaName(t, kubeOptions, crdIdentity.ClusterName) {
+
+		isCRDDeleted := deleteCustomResourceDefinition(t, kubeOptions, crdIdentity.ClusterName, timeoutSeconds)
+		if !isCRDDeleted {
+			Log(t, "CRD NOT deleted.", "deleteCustomResourceDefinition", "", nil)
+			return false
+		}
+	}
+	return true
+}
+
+// Log helper for output of action details.
+func Log(t *testing.T, subject string, action string, result string, errorDetail error) {
+
+	if errorDetail != nil {
+		logger.Log(t, fmt.Sprintf("[ %s ] [ %s] [ Error: %s]", subject, action, errorDetail.Error()))
+	} else {
+		if result != "" {
+			logger.Log(t, fmt.Sprintf("[ %s ] [ %s ] [ %s]", subject, action, result))
+		} else {
+			logger.Log(t, fmt.Sprintf("[ %s ] [ %s ]", subject, action))
+		}
+	}
+}
+
+// LogResults helper for output of map-based result details
+func LogResults(t *testing.T, message string, results map[string]string) {
+	for k, v := range results {
+		logger.Log(t, fmt.Sprintf("%s Id: %s Result: %s", message, k, v))
+	}
+}
+
+// Pause helper responsible for pausing for a specified time duration and interval.
+func Pause(t *testing.T, message string, interval, timeout time.Duration) {
```

Comment:
Out of interest.. what's wrong with ``time.Sleep(time.Duration)`` ? It blocks the currently executing goroutine as well.

Reply:
Right, nothing wrong with that usage as well.  This was a **util/helper function** in our test util module that assists with using a one-liner in our test setup/teardown where we want to identify why the pause, with a bit more flexibility for interval as needed.  I found that this logging context really helped when sifting through the log details when something isn't working as expected.

An example:
`Pause(t, "Pause for Cleanup Resource Completion", 1, 8*time.Second)`